src/util.coffee

```diff
@@ -32,3 +32,25 @@
                                       node?.tag is 'tag:yaml.org,2002:map'
 
 @NON_PRINTABLE = /[^\x09\x0A\x0D\x20-\x7E\x85\xA0-\uD7FF\uE000-\uFFFD]/
+
+@stringifyJSONCircularDepth = (json, depth = 1000) ->
```

Comment:
> I wonder when would a cycle ever be broken, why would counting to 500 be more effective than counting to 2?

The depth is the max depth of a circular reference, you can check both files:
https://github.com/repocho/raml-js-parser/blob/deb2ef9ec8b2a797a6e4d7f17267f169b07c1782/test/specs/dereferenceCircularDepth1.js
https://github.com/repocho/raml-js-parser/blob/deb2ef9ec8b2a797a6e4d7f17267f169b07c1782/test/specs/dereferenceCircular.js
And check that the generated schema have an aditional level depth with `maxDepth=1` .

_About the includes you meant in the raml definition ?_ The objective of this PR is to avoid the circular exception of JSON.stringify of a dereference schema definition with circular dependencies of $ref.